### ISSUE TITLE
Documentation missing/misleading on "Adding authentication providers"

### ISSUE THREAD

**Issue by RonnyA** [backstage/backstage#9055]
I am trying to add a new auth provider to single-sign in using EasyAuth as the BackStage app is running protected in Azure App service.

(And I am no JS/Node/Express expert)

This is the documentation I am following: https://backstage.io/docs/auth/add-auth-provider

## Expected Behavior

After running the documented commands I am expecting some code to be installed.

>cd plugins/auth-backend
>yarn add passport-provider-a
>yarn add @types/passport-provider-a

## Current Behavior

Folder "plugins/auth-backend" doesn't exists
if I do create the folder manually and run the Yarn commands I get

----
yarn add v1.22.17
error Running this command will add the dependency to the workspace root rather than the workspace itself, which might not be what you want - if you really meant it, make it explicit by running this command again with the -W flag (or --ignore-workspace-root-check).

info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
----

## Possible Solution

I am guessing I need to install the  plugin-auth-backend first. 

However there is no documentation on how to at  https://github.com/backstage/backstage/tree/master/plugins/auth-backend

But maybe there is a better approach to my challenge:

I am trying to make BackStage read the JSWT token from the existing Azure AppService endpoint "./auth/me" - 
This is to replace the "guest" login.

## Your Environment

Backstage 0.4.12

The starting point is an empty/default Backstage app after I ran "npx @backstage/create-app"

- NodeJS Version (v14):  Node 14 and 16 tried
- Operating System and Version (e.g. Ubuntu 14.04): Windows WSL/Ubuntu 20.04
- Browser Information:

**Comment by freben** [comment]
That guide is geared to contributions to the main repo. So for example, us maintainers and others that make open source contributions. This means that you'd be expected to first check out the main repo at `https://github.com/backstage/backstage` and contributing to it.

**Comment by Rugvip** [comment]
Yep that documentation is indeed quite misleading. Will piggyback on this issue as a tracker for getting that fixed.

I think the initial fix would be to clarify that those docs are for adding a provider to the main repo.

A possible followup would be to add a section to those docs that describe how to create a new provider in your own repo.

**Comment by jbadeau** [comment]
Just ran into this issue myself. I will extend auth-backend plugin for now but a guide on adding non oauth passport providers in a separate repo would be much appreciated.

**Comment by sieuvb** [comment]
Hi @freben, I have no idea how to hook the auth-backend plugin to the backend according to the [article](https://backstage.io/docs/auth/add-auth-provider#hook-it-up-to-the-backend), it's quite ambiguous to me. Could you please detail the guide or provide a simple example?

**Comment by jn-github** [comment]
> Hi @freben, I have no idea how to hook the auth-backend plugin to the backend according to the [article](https://backstage.io/docs/auth/add-auth-provider#hook-it-up-to-the-backend), it's quite ambiguous to me. Could you please detail the guide or provide a simple example?

@sieuvb, @jbadeau, @freben, I am looking help with oauth2 plugin implementation in my backstage app. any help would be appreciated.
